### ISSUE TITLE
Stratification during analysis

### ISSUE THREAD

**Issue by vlad-arista** [google/mangle#21]
Is there a reason why stratification is done during evaluation rather than analysis? It seems redundant to stratify every time given that analysis.ProgramInfo is most commonly produced by Analyze* calls.

**Comment by burakemir** [comment]
I agree that needless work is done when one has does multiple evaluation phases with the same program. I will add entry points that let clients call evaluation with a previously stratified program.

Also, stratification can be considered a form of semantic analysis, although it seems different from name resolution and type checking.

Are you suggesting that analysis should always perform stratification and include the result in analysis.ProgramInfo?
